src/process/main.cpp

```diff
@@ -87,6 +89,50 @@ int main( int argc, char *argv[] )
   }
   QgsApplication::setPrefixPath( myPrefixPath, true );
 
+  // Set translation context for qgis_process: necessary for naming of additional fields created by algorithms
+  // The command line parameter evaluation is done in qgisprocessing.cpp,
+  // which is too late for setting the language context.
+
+  QString translationCode;
+  int i = 1;
+  for(; i < argc; ++i)
+    if ( i < argc && ( argv[i] == QLatin1String( "--lang" ) || argv[i] == QLatin1String( "-l" ) ) )
+    {
+        translationCode = argv[++i];
+    }
```

Comment:
There's a potential crash here if "--lang" is specified with no following values:

```suggestion
  for( int i = 1; i < argc - 1; ++i)
  {
    if ( argv[i] == QLatin1String( "--lang" ) || argv[i] == QLatin1String( "-l" ) )
    {
        translationCode = argv[++i];
        break;
    }
  }
```